### PR TITLE
Apply Datadog privacy attributes to Forms Library review components

### PR DESCRIPTION
**Pull request by longmd**
## Summary
As teams have been setting up Datadog's RUM (Real User Monitoring), it was discovered that PII could be exposed on the Review Page during RUM session recordings because there are no privacy attributes applied to the review widgets. This PR applies the necessary attributes to allow the form data to be masked during RUM session recordings and for the action panel.

## Related issue(s)
department-of-veterans-affairs/va.gov-team#72280

## Acceptance criteria

- No PII will be exposed from the dynamic review widgets

### Quality Assurance & Testing

- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### CODE REVIEW

File: src/platform/forms-system/src/js/review/SSNWidget.jsx [hunk 1]
```diff
@@ -3,11 +3,21 @@ import React from 'react';
...
   if (value && value.length === 9) {
     return (
-      <span>{`${value.substr(0, 3)}-${value.substr(3, 2)}-${value.substr(
-        5,
-      )}`}</span>
+      <span
+        className="dd-privacy-hidden"
+        data-dd-action-name="Social Security number"
+      >
+        {`${value.substr(0, 3)}-${value.substr(3, 2)}-${value.substr(5)}`}
```
Comment (Mottie) [inline src/platform/forms-system/src/js/review/SSNWidget.jsx:10]: [`substr` is deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr); we can replace with [`substring`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring).

File: src/platform/forms-system/src/js/review/widgets.jsx [hunk 2]
```diff
@@ -2,11 +2,19 @@ import React from 'react';
...
+    <span className="dd-privacy-hidden" data-dd-action-name="data value">
+      {value}
+    </span>
+  );
 }
 
 export function DateWidget({ value, options }) {
-  return <span>{formatReviewDate(value, options.monthYear)}</span>;
+  return (
+    <span className="dd-privacy-hidden" data-dd-action-name="data value">
```
Comment (Mottie) [inline src/platform/forms-system/src/js/review/widgets.jsx:14]: "date value"?

File: src/platform/forms-system/src/js/review/widgets.jsx [hunk 3]
```diff
@@ -36,11 +48,15 @@ export const yesNo = ({ value, options = {} }) => {
     displayValue = labels.N || 'No';
   }
 
-  return <span>{displayValue}</span>;
+  return (
+    <span className="dd-privacy-hidden" data-dd-action-name="data value">
```
Comment (Mottie) [inline src/platform/forms-system/src/js/review/widgets.jsx:52]: Do we need to hide yes/no values? If yes, then maybe an action name of "yes or no value" works better?

**Review by Mottie — APPROVED** [review]

**Review by powellkerry — APPROVED** [review]

File: src/platform/forms-system/src/js/review/SSNWidget.jsx (hunk 1 above)
Comment (longmd) [reply src/platform/forms-system/src/js/review/SSNWidget.jsx:10]: I'd rather that be part of a larger effort to update things like that. I don't want this to be a "while I'm in here"-type PR.

**Pull request merged**